nominal/experimental/stream_v2/_write_stream.py

```diff
@@ -136,16 +165,74 @@ def _write_serialized_batch(
     pool: ThreadPoolExecutor,
     clients: WriteStreamV2._Clients,
     nominal_data_source_rid: str,
-    future: concurrent.futures.Future[bytes],
+    item_queue: Queue[BatchItem | QueueShutdown],
+    track_metrics: bool,
+    future: concurrent.futures.Future[tuple[bytes, IntegralNanosecondsUTC, IntegralNanosecondsUTC]],
 ) -> None:
     try:
-        serialized_data = future.result()
-        pool.submit(
-            clients.proto_write.write_nominal_batches,
+        serialized_data, oldest_timestamp, newest_timestamp = future.result()
+        write_future = pool.submit(
+            clients.proto_write.write_nominal_batches_with_metrics,
             clients.auth_header,
             nominal_data_source_rid,
             serialized_data,
+            oldest_timestamp,
+            newest_timestamp,
         )
+
+        def on_write_complete(
+            f: concurrent.futures.Future[tuple[float, float, float, float, float]],
+        ) -> None:
+            try:
+                (
+                    oldest_timestamp_diff_in_batch_before_request,
+                    newest_timestamp_diff_in_batch_before_request,
+                    rtt,
+                    largest_e2e_rtt,
+                    smallest_e2e_rtt,
+                ) = f.result()  # Check for exceptions
+
+                current_time_ns = int(datetime.now(timezone.utc).timestamp() * 1e9)
+                item_queue.put(
+                    BatchItem(
+                        channel_name="metric.oldest_timestamp_diff_in_batch_before_request",
```

Comment:
preference to use like a double underscore or some other way to disambiguate it from customer data